### PR TITLE
Final 1.10 bug bash

### PR DESCRIPTION
**Pull request by luxaritas**

### CODE REVIEW

File: src/eterna/mode/DesignBrowser/DesignBrowserMode.ts [hunk 1]
```diff
@@ -614,75 +612,49 @@ export default class DesignBrowserMode extends GameMode {
...
+                    this.getSolutionAtIndex(Math.max(0, this._currentSolutionIndex - 1))
+                );
+            },
             onNext: () => {
                 const nextSolutionIndex = Math.min(
                     this._filteredSolutions.length - 1,
                     this._currentSolutionIndex + 1
                 );
-                switchSolution(nextSolutionIndex);
+                this.showSolutionDetailsDialog(this.getSolutionAtIndex(nextSolutionIndex));
```
Comment (everyday847) [inline src/eterna/mode/DesignBrowser/DesignBrowserMode.ts:639]: this seems like a more sustainable solution anyway

File: src/eterna/mode/DesignBrowser/DesignBrowserMode.ts [hunk 2]
```diff
@@ -1023,12 +995,7 @@ export default class DesignBrowserMode extends GameMode {
     }
 
     private getSolutionIndex(solutionID: number): number {
-        for (let ii = 0; ii < this._filteredSolutions.length; ii++) {
-            if (this._filteredSolutions[ii].nodeID === solutionID) {
-                return ii;
-            }
-        }
-        return -1;
+        return this._filteredSolutions.findIndex((solution) => solution.nodeID === solutionID);
```
Comment (everyday847) [inline src/eterna/mode/DesignBrowser/DesignBrowserMode.ts:998]: naturally we had reimplemented a builtin

File: src/eterna/mode/FeedbackViewMode.ts [hunk 3]
```diff
@@ -56,14 +56,6 @@ export default class FeedbackViewMode extends GameMode {
         const background = new Background();
         this.addObject(background, this.bgLayer);
 
-        // this._puzzleTitle = Fonts.std(this._puzzle.getName(false), 14).color(0xffffff).bold().build();
-        // this._puzzleTitle.position = new Point(33, 8);
-        // this.uiLayer.addChild(this._puzzleTitle);
-
-        this._title = Fonts.std('', 12).color(0xffffff).bold().build();
```
Comment (everyday847) [inline src/eterna/mode/FeedbackViewMode.ts:63]: do we not want this anymore, or were we never really using it?

File: src/eterna/mode/FeedbackViewMode.ts [hunk 4]
```diff
@@ -73,22 +65,21 @@ export default class FeedbackViewMode extends GameMode {
...
         this.addObject(this._homeButton, this.uiLayer);
 
         const homeArrow = new Sprite(BitmapManager.getBitmap(Bitmaps.ImgHomeArrow));
         homeArrow.position = new Point(45, 14);
-        Assert.assertIsDefined(this.container);
-        this.container.addChild(homeArrow);
+        this.uiLayer.addChild(homeArrow);
 
-        const puzzleTitle = UITheme.makeTitle(this._puzzle.getName(true), 0xffffff);
+        const puzzleTitle = UITheme.makeTitle(this._puzzle.getName(!Eterna.MOBILE_APP), 0xffffff);
```
Comment (everyday847) [inline src/eterna/mode/FeedbackViewMode.ts:77]: aha, I assume this is related. we didn't need it after all

File: src/eterna/ui/Toolbar.ts [hunk 5]
```diff
@@ -386,57 +386,63 @@ export default class Toolbar extends ContainerObject {
...
 
         this.downloadSVGButton = new GameButton()
             .allStates(Bitmaps.CustomLayout)
             .disabled(undefined)
             .label('Download SVG format', 14)
             .scaleBitmapToLabel()
             .tooltip('Download an SVG of the current RNA layout');
-        this.actionMenu.addSubMenuButton(alterMenuIdx, this.downloadSVGButton);
+
+        if (this._boostersData != null && this._boostersData.actions != null) {
```
Comment (everyday847) [inline src/eterna/ui/Toolbar.ts:431]: I don't love it, but I accept it.

**Review by everyday847 — APPROVED** [review]
Phew.

File: src/eterna/mode/FeedbackViewMode.ts (hunk 3 above)
Comment (luxaritas) [reply src/eterna/mode/FeedbackViewMode.ts:63]: Certainly the former, if not the latter

File: src/eterna/mode/FeedbackViewMode.ts (hunk 4 above)
Comment (luxaritas) [reply src/eterna/mode/FeedbackViewMode.ts:77]: This is actually a change I didn't realize was made due to a copy-paste, but if we ever implement this into the mobile app (who knows!) it'll be necessary

Comment (luxaritas) [reply src/eterna/mode/FeedbackViewMode.ts:77]: Oh, I see you were referring to the previous comment - yup

File: src/eterna/ui/Toolbar.ts (hunk 5 above)
Comment (luxaritas) [reply src/eterna/ui/Toolbar.ts]: Changed

**Review by everyday847 — APPROVED** [review]
Excellent, and nice to know that now we're actually covering the `Progression` type.

**Pull request merged**